### PR TITLE
ZCS-12009 : Removed perl script and added util method for sending notification mail

### PR DESCRIPTION
**Pull request by abhishekdeshmukh2209**
Issue :- Unable to send mail via perl script for u20, having issue.

Fix :- Added new utility method for sending email
          Removed older API

### CODE REVIEW

File: store/src/java/com/zimbra/cs/datasource/SyncUtil.java [hunk 1]
```diff
@@ -121,4 +133,65 @@ public static Log getTraceLogger(Log parent, String id) {
         return log;
     }
 
+    public static boolean sendReportEmail(String subject, String text, String[] toStrs) {
+        String fromStr;
+        try {
+            Provisioning prov = Provisioning.getInstance();
+            Server server = prov.getLocalServer();
+            if (toStrs == null || toStrs.length == 0)
```
Comment (dasiyogesh) [inline store/src/java/com/zimbra/cs/datasource/SyncUtil.java:141]: use curly braces for if block

File: store/src/java/com/zimbra/cs/datasource/SyncUtil.java [hunk 2]
```diff
@@ -121,4 +133,65 @@ public static Log getTraceLogger(Log parent, String id) {
...
+                return false;
+            fromStr = server.getAttr(Provisioning.A_zimbraBackupReportEmailSender);
+            if (fromStr == null)
```
Comment (dasiyogesh) [inline store/src/java/com/zimbra/cs/datasource/SyncUtil.java:144]: use curly braces for if block

File: store/src/java/com/zimbra/cs/service/admin/SendMdmNotificationEmail.java [hunk 3]
```diff
@@ -1,70 +0,0 @@
...
-    @Override
-    public Element handle(Element request, Map<String, Object> context) throws ServiceException {
-        ZimbraSoapContext zsc = getZimbraSoapContext(context);
-        Provisioning prov = Provisioning.getInstance();
-        Server localServer = prov.getLocalServer();
-        AccountUtil.isAdminAccount(getAuthenticatedAccount(zsc));
-        String message = request.getElement(SyncAdminConstants.E_MESSAGE).getText();
-        String status = request.getElement(SyncAdminConstants.E_STATUS).getText();
-        RemoteManager remoteManager = RemoteManager.getRemoteManager(prov.getLocalServer());
-        String command = "zmmdmnotificationmail \"" + message + "\" \"" + status + "\"";
```
Comment (silentsakky) [inline store/src/java/com/zimbra/cs/service/admin/SendMdmNotificationEmail.java:48]: we are going to remove `zmmdmnotificationmail` also right?

**Review by dasiyogesh — APPROVED** [review]

File: store/src/java/com/zimbra/cs/service/admin/SendMdmNotificationEmail.java (hunk 3 above)
Comment (dasiyogesh) [reply store/src/java/com/zimbra/cs/service/admin/SendMdmNotificationEmail.java:48]: should be not required anymore

Comment (silentsakky) [reply store/src/java/com/zimbra/cs/service/admin/SendMdmNotificationEmail.java:48]: yes so we should remove it from zm-build and zm-core-utils

**Review by deepakzimb — APPROVED** [review]

File: store/src/java/com/zimbra/cs/datasource/SyncUtil.java (hunk 1 above)
Comment (abhishekdeshmukh2209) [reply store/src/java/com/zimbra/cs/datasource/SyncUtil.java]: done

File: store/src/java/com/zimbra/cs/datasource/SyncUtil.java (hunk 2 above)
Comment (abhishekdeshmukh2209) [reply store/src/java/com/zimbra/cs/datasource/SyncUtil.java]: done

File: store/src/java/com/zimbra/cs/service/admin/SendMdmNotificationEmail.java (hunk 3 above)
Comment (abhishekdeshmukh2209) [reply store/src/java/com/zimbra/cs/service/admin/SendMdmNotificationEmail.java:48]: done
https://github.com/Zimbra/zm-build/pull/233
https://github.com/Zimbra/zm-core-utils/pull/124

**Pull request merged**